object/slicer/slicer.go

```diff
@@ -0,0 +1,255 @@
+package slicer
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"hash"
+	"io"
+
+	"github.com/nspcc-dev/neofs-sdk-go/checksum"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/version"
+	"github.com/nspcc-dev/tzhash/tz"
+)
+
+// ObjectWriter represents a virtual object recorder.
+type ObjectWriter interface {
+	// InitDataStream initializes and returns a stream of writable data associated
+	// with the object according to its header. Provided header includes at least
+	// owner, container and object ID fields.
+	InitDataStream(header object.Object) (dataStream io.Writer, err error)
+}
+
+// SliceDataIntoObjects creates new NeoFS object from the input data stream,
+// associates the object with the given container, makes it owned by the
+// specified user, and writes the object to the given ObjectWriter. After a
+// successful write, SliceDataIntoObjects returns an oid.ID which is a unique
+// reference to the object in the container.
+//
+// SliceDataIntoObjects sets all required calculated fields like payload length,
+// checksum, etc.
+//
+// If ObjectWriter provides io.Closer, it is closed after the payload has been
+// written. In this case, SliceDataIntoObjects fails immediately on Close error.
+//
+// By default, single object with empy payload is created. Various Options allow
```

Comment:
Not a useful default.

Reply:
Follows from https://github.com/nspcc-dev/neofs-sdk-go/pull/382#discussion_r1119285703.